File: Web/personal-site/src/app/models/project.ts
import { Skill } from './skill';
import { Technology } from './technology';

export class Project {
    id:           number;
    userId:       number;
    title:        string;
    description:  string;
    githubRepoId: null;
    projectUrl:   string;
    projectType:  string;
    githubRepo:   null;
    user:         null;
    skills:       Skill[];
    technologies: Technology[];

    public constructor(init?:Partial<Project>) {
        Object.assign(this, init);
    }
}

File: Shared/Site.Application/Entities/Project.cs
using System.Collections.Generic;

namespace Site.Application.Entities
{
    public class Project
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int? GithubRepoId { get; set; }
        //public int ProjectTypeId { get; set; }
        public string ProjectUrl { get; set; }
        public Enums.ProjectType ProjectType { get; set; }
        public GithubRepo GithubRepo { get; set; }
        public User User { get; set; }
        public virtual ICollection<ProjectSkill> ProjectSkills { get; set; }
        public virtual ICollection<ProjectTechnology> ProjectTechnologies { get; set; }
    }
}